Shoko.Server/API/v3/Controllers/PlexController.cs

```diff
@@ -0,0 +1,203 @@
+    /// Check if the current user is currently authenticated against the plex api.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("Auth/IsAuthenticated")]
+    public ActionResult<bool> IsAuthenticated()
+    {
+        var helper = GetHelperForUser();
+
+        return helper.IsAuthenticated;
+    }
+
+    /// <summary>
+    /// Invalidate and remove the current plex authentication token.
+    /// </summary>
+    /// <returns></returns>
+    [HttpPost("Auth/InvalidateToken")]
+    public ActionResult<bool> InvalidateToken()
+    {
+        var helper = GetHelperForUser();
+
+        if (!helper.IsAuthenticated)
+            return false;
+
+        helper.InvalidateToken();
+
+        return true;
+    }
+
+    /// <summary>
+    /// Show all available server for the current user.
+    /// </summary>
+    /// <param name="userId"></param>
+    /// <returns></returns>
+    [HttpGet("Linking/AvailableServers")]
+    public ActionResult<List<PlexServer>> AvailableServers(int userId)
+    {
+        var helper = GetHelperForUser();
+
+        if (!helper.IsAuthenticated)
+            return BadRequest(NotAuthenticated);
+
+        var currentServerID = helper.ServerCache?.ClientIdentifier ?? "";
+        return helper.GetPlexServers()
+            .Select(device => new PlexServer(device, device.ClientIdentifier == currentServerID))
+            .ToList();
+    }
+
+    /// <summary>
+    /// Get the active server.
+    /// </summary>
+    /// <returns></returns>
+    [HttpGet("Linking/Server")]
+    public ActionResult<PlexServer> CurrentDevice()
+    {
+        var helper = GetHelperForUser();
+
+        if (!helper.IsAuthenticated)
+            return BadRequest(NotAuthenticated);
+
+        var currentServer = helper.ServerCache;
+        if (currentServer == null)
+            return BadRequest(NoSelectedServer);
+
+        return new PlexServer(currentServer, true);
+    }
+
+    /// <summary>
+    /// Select the active server.
+    /// </summary>
+    /// <param name="server"></param>
+    /// <returns></returns>
+    [HttpPut("Linking/Server")]
+    public ActionResult<PlexServer> UseDevice(PlexServer server)
+    {
+        var helper = GetHelperForUser();
+
+        if (!helper.IsAuthenticated)
+            return BadRequest(NotAuthenticated);
+
+        var previousServer = helper.ServerCache;
+        var currentServer = helper.GetPlexServers()
+            .FirstOrDefault(ser => ser.ClientIdentifier == server.ID);
+
+        if (currentServer == null || !currentServer.Provides.Split(',').Contains("server"))
+            return BadRequest(InvalidServerSelection);
+
+        helper.UseServer(currentServer);
+
+        // Reset the selected libraries if we're switching server.
+        if (previousServer != null && currentServer.ClientIdentifier != previousServer.ClientIdentifier)
+        {
+            ServerSettings.Instance.Plex.Libraries = new();
+            ServerSettings.Instance.SaveSettings();
+        }
```

Comment:
Move this to the `helper.UseServer()` call

The reason for this means if we add other places/methods to change the server it means the expected patterns are the same.